### ISSUE TITLE
Adding claims to request

### ISSUE THREAD

**Issue by davidkeaveny** [webmotions/fake-authentication-jwtbearer#14]
Does anyone know how to add a claim to the token being generated, if the claim type is essentially a URI? I'm working on an application that integrates with the Auth0 service and uses custom claims. Auth0 requires that custom claims are prefixed with a URI (presumably to guarantee uniqueness), so my claim might be something like:

```
var claim = new System.Security.Claim("http://mycompany.com/customClaim", "someValue");
```

I have no problem creating claims that have a more simple syntax for the name e.g.

```
dynamic token = new ExpandoObject();
token.customClaim = "someValue";
client.SetFakeBearerToken("test.user@acme.inc", new[] { "admin" }, (object) token);
```

If I try the following, then I get an exception:

```
var claims = new Dictionary<string, object>
{
    { ClaimTypes.Name, "test.user@acme.inc" },
    { ClaimTypes.Role, "admin" },
    { "http://mycompany.com/customClaim", "someValue" },
};

client.SetFakeJwtBearerToken(claims);
```

where the exception is ` 'e' is an invalid start of a value. Path: $ | LineNumber: 0 | BytePositionInLine: 0`. Checking the test library, I see that the `BearerValueType` needs to be set to `FakeJwtBearerBearerValueType.Jwt` to make this error go away, which it does:

```
services.AddAuthentication(options =>
{
    options.DefaultAuthenticateScheme = FakeJwtBearerDefaults.AuthenticationScheme;
    options.DefaultChallengeScheme = FakeJwtBearerDefaults.AuthenticationScheme;
}).AddFakeJwtBearer(options =>
{
    options.BearerValueType = FakeJwtBearerBearerValueType.Jwt;

    var handler = (FakeJwtBearerClaimsHandler)options.SecurityTokenClaimHandler;
    handler.Options.Issuer = "https://mydomain.auth0.com/";
    handler.Options.OriginalIssuer = "https://mydomain.auth0.com/";
    handler.Options.OverrideIssuer = true;
})
```

However, the code to change the issuer then no longer works. So, I can either have vanilla claims with a custom issuer, or custom claims with the default issuer; what I really want is custom claims with a custom issuer. Any thoughts on what needs to be done?

**Comment by DOMZE** [comment]
Hello,
Your assumptions are correct.

BearerValueType with value `FakeJwtBearerBearerValueType.Jwt` should use the HttpClient extension method `SetFakeJwtBearerToken`

BearerValueType with value `FakeJwtBearerBearerValueType.Json` should use any of the HttpClient extension method `SetFakeBearerToken`

The issuer overriding is not handled with the `FakeJwtBearerBearerValueType.Jwt` (as I assumed you can create your JWT and set your claims from there, manually).

I agree there's an incoherence between the Base64/Json and Jwt with the overriding of issuer. I will make the proper changes for everything to be coherent.

For now what you can do is remove the Issuer options and create your JWT using the BearerValueType with value `FakeJwtBearerBearerValueType.Jwt` and the HttpClient extension method `SetFakeJwtBearerToken`

**Comment by DOMZE** [comment]
Fixed in 6.1.0

**Comment by davidkeaveny** [comment]
Thanks @DOMZE I can confirm that that fixed things nicely for me!
